consensus/src/config.rs

```diff
@@ -100,25 +96,47 @@ pub fn exponential_slowdown(
     Duration::from_millis(delay)
 }
 
-/// A default configuration of what the creators of this package see as optimal parameters.
-pub fn default_config(n_members: NodeCount, node_ix: NodeIndex, session_id: SessionId) -> Config {
-    Config {
+/// Creates a [`Config`], allowing the user to omit specifying `max_round` and `delay_config`
+/// in which case they will be set do default values, suggested by the creators of this package.
+/// `time_to_reach_max_round` is a lower bound on the time needed to reach the maximum round
+/// expected by the user and is only used for verification.
+pub fn create_config(
+    n_members: NodeCount,
+    node_ix: NodeIndex,
+    session_id: SessionId,
+    max_round: Option<Round>,
+    delay_config: Option<DelayConfig>,
+    time_to_reach_max_round: Duration,
+) -> Result<Config, InvalidConfigError> {
+    let max_round = max_round.unwrap_or(5000);
+
+    let delay_config = delay_config.unwrap_or(DelayConfig {
```

Comment:
Is this really the easiest way of creating a `DelayConfig`?

Reply:
It was initially done like that, and I haven't seen any function like `default_delay_config` or so. But we can consider wrapping this code in such a function.

Reply:
Yeah, it would be nice for the creation of delay configs to be better designed, especially since we want to do that more manually now. Some sensible constructor(s?) for it, and perhaps even for various pieces of it.